### PR TITLE
dbeaver/pro#136 add filter appending to user query for Influx

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/pro#136

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 1]
```diff
@@ -103,6 +106,38 @@ public void appendConditionString(
         appendConditionString(filter, constraints, dataSource, conditionTable, query, inlineCriteria, subQuery);
     }
 
+    @Override
+    public String addFiltersToQuery(
+        @Nullable DBRProgressMonitor monitor,
+        final @NotNull DBPDataSource dataSource,
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: IMHO this looks better
```suggestion
        @NotNull final DBPDataSource dataSource,
```

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 2]
```diff
@@ -103,6 +106,38 @@ public void appendConditionString(
...
+        @NotNull String sqlQuery,
+        final @NotNull DBDDataFilter dataFilter
+    ) {
+        return dataSource.getSQLDialect().getQueryGenerator().addFiltersToQuery(monitor, dataSource, sqlQuery, dataFilter);
+    }
+
+
+    @Override
+    public String wrapQuery(@NotNull DBPDataSource dataSource, @NotNull String sqlQuery,
+                                   @NotNull DBDDataFilter dataFilter) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: I'm not sure about the reason to wrap one argument to **the middle** of the next line, but I'd format it this way.
```suggestion
    public String wrapQuery(
        @NotNull DBPDataSource dataSource,
        @NotNull String sqlQuery,
        @NotNull DBDDataFilter dataFilter
    ) {
```

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: It's not required at all TBH

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java [hunk 3]
```diff
@@ -88,6 +89,31 @@ void appendConditionString(@NotNull DBDDataFilter filter,
...
+     * Applying filters changes query formatting (thus it changes column names in expressions)
+     * Solution - always wrap query in subselect + add patched WHERE and ORDER
+     * It is configurable
+     * @param monitor database progress monitor
+     * @param dataSource datasource
+     * @param sqlQuery user query
+     * @param dataFilter filter conditions
+     * @return modified query
+     */
+    String addFiltersToQuery(
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java]: This method's name is confusing. It says that it `add`s something to something, but it returns `String`.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java [hunk 4]
```diff
@@ -88,6 +89,31 @@ void appendConditionString(@NotNull DBDDataFilter filter,
...
+    );
+
+    /**
+     * Appends conditions to the user Query
+     * @param dataSource datasource
+     * @param sqlQuery user query
+     * @param dataFilter filter conditions
+     * @return modified query
+     */
+    String wrapQuery(@NotNull DBPDataSource dataSource, @NotNull String sqlQuery, @NotNull DBDDataFilter dataFilter);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java]: This method's name is confusing either. What does it mean to `wrap` a query? Please find a more suitable name.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 5]
```diff
@@ -149,7 +152,7 @@ public void appendConditionString(
                         // Seems to a reference on a table column.
                         // It is better to use real table column in expressions because aliases may not work
                         attrName = DBUtils.getQuotedIdentifier(dataSource,
-                            subQuery ? constraint.getAttributeLabel() : constraint.getAttributeName());
+                                                               subQuery ? constraint.getAttributeLabel() : constraint.getAttributeName());
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: ?

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 6]
```diff
@@ -178,6 +181,39 @@ public void appendConditionString(
...
+    }
+
+
+    @Override
+    public String wrapQuery(
+        @NotNull DBPDataSource dataSource,
+        @NotNull String sqlQuery,
+        @NotNull DBDDataFilter dataFilter
+    ) {
+        // Append filter conditions to query
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: It should be obvious from the method itself, not from some comment between its code

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 7]
```diff
@@ -149,7 +151,10 @@ public void appendConditionString(
                         // Seems to a reference on a table column.
                         // It is better to use real table column in expressions because aliases may not work
                         attrName = DBUtils.getQuotedIdentifier(dataSource,
-                            subQuery ? constraint.getAttributeLabel() : constraint.getAttributeName());
+                                                               subQuery
+                                                               ? constraint.getAttributeLabel()
+                                                               : constraint.getAttributeName()
+                        );
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: ???

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java [hunk 8]
```diff
@@ -88,6 +89,29 @@ void appendConditionString(@NotNull DBDDataFilter filter,
...
 
+    /**
+     * Applies filters to the existing user queries
+     * @param monitor database progress monitor
+     * @param dataSource datasource
+     * @param sqlQuery user query
+     * @param dataFilter filter conditions
+     * @return modified query
+     */
+    String getQueryWithAppliedFilters(
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java:120]: Please specify nullability annotation for the return type

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java [hunk 9]
```diff
@@ -88,6 +89,29 @@ void appendConditionString(@NotNull DBDDataFilter filter,
...
+    );
+
+    /**
+     * returns user query with filter and order
+     * @param dataSource datasource
+     * @param sqlQuery user query
+     * @param dataFilter filter conditions
+     * @return modified query
+     */
+    String getWrappedFilterQuery(@NotNull DBPDataSource dataSource, @NotNull String sqlQuery, @NotNull DBDDataFilter dataFilter);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java:136]: Please specify nullability annotation for the return type

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 10]
```diff
@@ -178,6 +181,38 @@ public void appendConditionString(
...
     }
 
+    @Override
+    public String getQueryWithAppliedFilters(
+        @Nullable DBRProgressMonitor monitor,
+        @NotNull DBPDataSource dataSource,
+        @NotNull String sqlQuery,
+        @NotNull DBDDataFilter dataFilter
+    ) {
+        return dataSource.getSQLDialect().getQueryGenerator().getQueryWithAppliedFilters(monitor, dataSource, sqlQuery, dataFilter);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java]: Looks weird, potential infinite loop

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java [hunk 11]
```diff
@@ -38,6 +41,7 @@
 
 public class StandardSQLDialectQueryGenerator implements SQLDialectQueryGenerator {
     private Log log = Log.getLog(StandardSQLDialectQueryGenerator.class);
+    private static final String NESTED_QUERY_AlIAS = "z_q";
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java:44]: What if you have nested query inside nested query?

**Review by E1izabeth — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java (hunk 11 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java:44]: Same as old logic, I just transferred it into the new class to be able to overload it.

**Review by E1izabeth — APPROVED** [review]

**Review by serge-rider — APPROVED** [review]

**Pull request merged**